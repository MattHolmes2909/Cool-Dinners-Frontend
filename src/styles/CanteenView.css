.CanteenView {
  margin: auto;
  text-align: center;
}

.order-message,
.tomorrow {
  display: inline-block;
  font-size: 1.5rem;
  padding: 0;
  margin: 0;
  margin-bottom: 1rem;
}

.tomorrow {
  color: #e9740e;
}

.table {
  border-collapse: collapse;
  margin: auto;
  width: 90%;
}

th {
  position: relative;
}

.foodicon {
  display: block;
  border: solid 6px;
  border-radius: 50%;
  margin: auto;
  width: 100px;
}

.meat {
  border-color: #ca4246;
}

.veg {
  border-color: #4cbf5d;
}

.fish {
  border-color: #5383e9;
}

.noDinner {
  border-color: #d2e367;
}

.schoolclass {
  background-color: #42ba78;
  font-size: 1.5rem;
  color: black;
}

.classRow:nth-of-type(2n) {
  background-color: #854345;
}
.classRow:nth-of-type(2n-1) {
  background-color: #9a4e4e;
}

.totalRow {
  background-color: #0cc25e;
  color: black;
  font-size: 1.2rem;
}

.totalRow > td:first-of-type {
  border-radius: 12px 0 0 0;
}

.totalRow > td:last-of-type {
  border-radius: 0 12px 0 0;
}

td {
  padding: 0.5rem 0;
}

.classRow > td:hover {
  animation: glowHover 0.5s forwards;
}

input {
  width: 100%;
}

.reset-button {
  background-color: #d2e367;
  border: 0;
  border-radius: 0 0 12px 12px;
  cursor: pointer;
  font-family: "Bebas Neue", sans-serif;
  font-size: 2rem;
  width: 90%;
}

.reset-button:hover {
  animation: resetButtonHover 0.5s forwards;
}

@keyframes resetButtonHover {
  to {
    background-color: #d7d71e;
  }
}
