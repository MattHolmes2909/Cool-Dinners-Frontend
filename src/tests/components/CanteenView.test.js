import React from "react";
import { render } from "@testing-library/react";
import CanteenView from "../../components/CanteenView";

describe("CanteenView", () => {
    it("renders correctly", () => {
        const { asFragment } = render (<CanteenView />);

        expect(asFragment()).toMatchSnapshot();
    });
});