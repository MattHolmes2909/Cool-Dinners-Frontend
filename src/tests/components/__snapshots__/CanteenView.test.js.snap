// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanteenView renders correctly 1`] = `
<DocumentFragment>
  <div
    class="CanteenView"
  >
    <p
      class="order-message"
    >
      You are ordering dinners forâ €
    </p>
    <p
      class="tomorrow"
    >
      Wednesday, August 11th 2021!
    </p>
    <form
      class="order-form"
    >
      <table
        class="table table-responsive"
      >
        <thead>
          <tr>
            <th
              name="className"
            />
            <th
              name="pizza"
            >
              <img
                alt="pizza"
                class="foodicon pizza meat"
                src="pizza.png"
              />
              Pizza
            </th>
            <th
              name="pasta"
            >
              <img
                alt="pasta"
                class="foodicon pasta veg"
                src="pasta.png"
              />
              Pasta
            </th>
            <th
              name="fish"
            >
              <img
                alt="fish"
                class="foodicon pizza fish"
                src="fish.png"
              />
              Fish and Chips
            </th>
            <th
              name="quorn"
            >
              <img
                alt="curry"
                class="foodicon pizza veg"
                src="curry.png"
              />
              Quorn Curry
            </th>
            <th
              name="noDinner"
            >
              <img
                alt="noDinner"
                class="foodicon noDinner"
                src="nodinner.png"
              />
              Dinner not needed
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="totalRow"
          >
            <td>
              <div
                class="radiotext"
              >
                <label
                  for="schoolclass"
                >
                  Total
                </label>
              </div>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
          </tr>
          <tr
            class="classRow"
          >
            <td>
              <div
                class="radiotext"
              >
                <label
                  for="schoolclass"
                >
                  1DS
                </label>
              </div>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
          </tr>
          <tr
            class="classRow"
          >
            <td>
              <div
                class="radiotext"
              >
                <label
                  for="schoolclass"
                >
                  1MH
                </label>
              </div>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
          </tr>
          <tr
            class="classRow"
          >
            <td>
              <div
                class="radiotext"
              >
                <label
                  for="schoolclass"
                >
                  2AW
                </label>
              </div>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
          </tr>
          <tr
            class="classRow"
          >
            <td>
              <div
                class="radiotext"
              >
                <label
                  for="schoolclass"
                >
                  2NM
                </label>
              </div>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </form>
  </div>
</DocumentFragment>
`;
